databricks/koalas/frame.py

```diff
@@ -1063,6 +1063,49 @@ def notnull(self):
 
     notna = notnull
 
+    def nunique(self, axis: int = 0, dropna: bool = True) -> 'ks.Series':
+        """
+        Return number of unique elements in the object.
+
+        Excludes NA values by default.
+
+        Parameters
+        ----------
+        axis : Can only be set to 0 at the moment.
+        dropna : Don’t include NaN in the count.
+
+        Returns
+        -------
+        The number of unique values per column as a Koalas Series.
+
+        Examples
+        --------
+        >>> ks.DataFrame({'A': [1, 2, 3], 'B': [np.nan, 3, np.nan]}).nunique()
+        A    3
+        B    1
+        Name: 0, dtype: int64
+
+        >>> ks.DataFrame({'A': [1, 2, 3], 'B': [np.nan, 3, np.nan]}).nunique(dropna=False)
+        A    3
+        B    2
+        Name: 0, dtype: int64
+        """
+        if axis != 0:
+            raise ValueError("The 'nunique' method only works with axis=0 at the moment")
+        if dropna:
+            res = self._sdf.select([F.countDistinct(Column(c))
+                                   .alias(c)
+                                    for c in self.columns])
+        else:
+            res = self._sdf.select([(F.countDistinct(Column(c))
+                                     + F.when(F.count(F.when(F.col(c).isNull(), 1).otherwise(None))
+                                              >= 1, 1).otherwise(0))
+                                   .alias(c)
+                                    for c in self.columns])
+        # TODO: Find a way to avoid converting to pandas and back
+        res = res.toPandas().T.iloc[:, 0]
+        return ks.Series(res)
```

Comment:
Also, I missed this: it should return a pandas series, because it is guaranteed to be small and is typically used for point indexing, see note in the contributing guide https://github.com/databricks/koalas#return-koalas-data-structure-for-big-data-and-pandas-data-structure-for-small-data

Reply:
Thanks for pointing out this section from the design principles. I believe this makes a strong case for using pandas Series instead of Koalas Series which I've done with 9c425a6. Also, this makes it somewhat more consistent with `Series.nunique()` which returns a simply Python int instead of some Spark-ish data type.